src/pal/src/loader/module.cpp

```diff
@@ -1620,6 +1620,8 @@ static HMODULE LOADLoadLibrary(LPCSTR shortAsciiName, BOOL fDynamic)
         shortAsciiName = FREEBSD_LIBC;
 #elif defined(__NetBSD__)
         shortAsciiName = "libc.so";
+#elif !HAVE_GNU_LIBNAMES_H
+        shortAsciiName = "libc.musl-x86_64.so";
```

Comment:
I have fixed it in a slightly different manner (keeping `LIBC_SO` as the last resort as it is now).